openfl-tutorials/interactive_api/jax_linear_regression/workspace/custom_adapter.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Custom model numpy adapter."""
+
+from openfl.plugins.frameworks_adapters.framework_adapter_interface import (
+    FrameworkAdapterPluginInterface,
+)
+
+
+class CustomFrameworkAdapter(FrameworkAdapterPluginInterface):
+    """Framework adapter plugin class."""
+
+    @staticmethod
+    def get_tensor_dict(model, optimizer=None):
+        """Extract tensors from a model."""
+        return {'w': model.weights}
```

Comment:
just for this regression example.

Reply:
Since you're adding a JAX example, your framework adapter should be able to handle other model architectures as well. For instance, a `pytorch` `model.state_dict()` can be iteratively mapped for each layer regardless of the architecture.

I'll let other reviewers chime in if this needs to be enforced or if they would like to archive it for future discussions. If latter, I'm ok and this can be closed.

Reply:
@acharyasunil  this should work for all JAX models. @MasterSkepticista  What is the rationale/usecase behind " your framework adapter should be able to handle other model architectures as well" ?  I didnt get the usecase behind pytorch model.state_dict()
Eventually tensor dicts are converted to numpy before sending to aggregator.

Reply:
It would be better to have a structured `get_tensor_dict` and `set_tensor_dict` with optimizer state for JAX based Neural Network example.
For the current regression example, a plain getter and setter for model.weights applicable for all JAX model is sufficient.